pkg/datasource/macros.go

```diff
@@ -0,0 +1,29 @@
+package datasource
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/grafana/influxdb-flux-datasource/pkg/models"
+)
+
+const timeFilter = `\$__timeFilter`
+
+// Interpolate processes macros
+func Interpolate(query models.QueryModel) (string, error) {
```

Comment:
Since flux is Go and os, depending on future plans, it might make sense to use flux/ast (or maybe flux/ast/edit ?) - use their parser.